sdk/nodejs/appservice/zMixins.ts

```diff
@@ -847,24 +847,28 @@ declare module "./functionApp" {
     }
 }
 
-FunctionApp.prototype.getHostKeys = function(this: FunctionApp) {
+FunctionApp.prototype.getHostKeys = function(this: FunctionApp, attempts?: number) {
     return this.id.apply(async id => {
+        const retryAttempts = attempts || 5
         const credentials = await core.getServiceClientCredentials();
         const client = new AzureServiceClient(credentials);
         const url = `https://management.azure.com${id}/host/default/listkeys?api-version=2018-02-01`;
 
         const response = await client.sendRequest({ method: "POST", url });
-        if (response.status >= 400) {
-            throw new Error(`Failed to retrieve the host keys: ${response.bodyAsText}`);
+        if(response.status < 400) {
```

Comment:
I find the previous flow easier to read, when we handle errors first, then the main branch is the success case. So, I would do 

```ts
if (response.status >= 400) {
    if (retryAttempts === 0) {
       // throw
    }
    // recurse
}

// parse the body and return
```

Reply:
Reverted to previous flow.